stix_shifter_utils/stix_translation/src/json_to_stix/json_to_stix_translator.py

```diff
@@ -190,169 +82,201 @@ def gen_dict_extract(self, key, var):
-                            for obj_name in object_map:
-                                if pattern.match(obj_name):
-                                    val = object_map.get(obj_name)
-                                    stix_value.append(val)
+                        return_value = self._get_tag_ind(references, object_tag_ref_map, create_on_absence=False)
+                        if unwrap is not False and not isinstance(return_value, list):
+                            return_value = [return_value]
+                else:
+                    if unwrap is False and observable_key and not self._check_stix_value_pattern(observable_key, value):
+                        return None
+                    return_value = value
+
+            return return_value
+        except Exception as e:
+            raise Exception("Error in json_to_stix_translator._compose_value_object: %s" % e)
+
+
+    def _get_tag_ind(self, tag, object_tag_ref_map, create_on_absence=False, unwrap=False, property_key=None):
+        tag_ind = None
+        if unwrap:
+            tag = tag + '_' + str(unwrap)
+
+        if tag in object_tag_ref_map['tags']:
+            tag_ind = object_tag_ref_map['tags'][tag]['i']
+            object_tag_ref_map['tags'][tag]['n'] += 1
+        elif create_on_absence:
+            tag_ind = object_tag_ref_map[UUID5_NAMESPACE]
+            object_tag_ref_map[UUID5_NAMESPACE] += 1
+            object_tag_ref_map['tags'][tag] = {'i': tag_ind, 'n': 0}
+            
+        if tag_ind is not None:
+            tag_ind = str(tag_ind)
+            if property_key and '_ref' not in property_key:
+                object_tag_ref_map['non_ref_props'][tag_ind] = True
+
+        return tag_ind
+
+    def _add_prperty(self, type_name, property_key, parent_key_ind, value, objects, group=False, cybox=True):
+        if not parent_key_ind in objects:
+            if cybox:
+                objects[parent_key_ind] = {
+                    'type': type_name,
+                    property_key: value
+                }
+            else:
+                objects[parent_key_ind] = {
+                property_key: value
+            }
+        else:
+            if not property_key in objects[parent_key_ind]:
+                objects[parent_key_ind][property_key] = value
+            elif isinstance(value, dict):
+                objects[parent_key_ind][property_key] = dict_merge(objects[parent_key_ind][property_key], value)
+            elif isinstance(objects[parent_key_ind][property_key], list) and group:
+                objects[parent_key_ind][property_key].extend(value)
+            else: #TODO: get rid of this weird 'else'
+                # print({'old': objects[parent_key_ind][property_key], 'new': value})
+                pass
+
+    def _process_properties(self, to_stix_config_prop, data, objects, object_tag_ref_map, parent_data=None, ds_sub_key=None, object_key_ind=None):
```

Comment:
you may wanna avoid using names like process. it may confuse people because process is a name of a stix object. I thought we you are just working on process properties :)

Reply:
agree, just didn't come up with more proper name. 
this function is kind of similar to the previous `_tranform` [here](https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_utils/stix_translation/src/json_to_stix/json_to_stix_translator.py#L209), 
but for some reason I did not like it, because there is "public" function `transform` already.